src/toil/common.py

```diff
@@ -796,13 +796,19 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         """
         Clean up after a workflow invocation. Depending on the configuration, delete the job store.
         """
-        try:
+        from toil.utils.toilStatus import ToilStatus
+        
+        try: 
             if (exc_type is not None and self.config.clean == "onError" or
                             exc_type is None and self.config.clean == "onSuccess" or
                         self.config.clean == "always"):
+
                 try:
-                    self._jobStore.destroy()
-                    logger.info("Successfully deleted the job store: %s" % str(self._jobStore))
+                    status = ToilStatus.getStatus(self._jobStore)
+                    
+                    if status == 'COMPLETED':
```

Comment:
Would we still want to destroy the job store if the user didn't call `start()`, since we've initialized a new job store when we enter the context manager? I feel like cleaning up only if the status is `COMPLETED` might create unused job stores if the user isn't calling `start()` either. 

Is it sufficient to just check if the restart option is set but `restart()` is not called at the top (before line 802)?